process/peer/process.go

```diff
@@ -345,6 +389,123 @@ func (p *validatorStatistics) getPeerAccount(address []byte) (state.PeerAccountH
 	return peerAccount, nil
 }
 
+// loadPreviousShardHeaders loads the previous shard headers for a given metablock. For the metachain it's easy
+//  since it has all the shard headers in its storage, but for the shard it's a bit trickier and we need
+//  to iterate through past metachain headers until we find all the ShardData we are interested in
+func (p *validatorStatistics) loadPreviousShardHeaders(currentHeader, previousHeader *block.MetaBlock) error {
+
+	p.mutPrevShardInfo.Lock()
+	defer p.mutPrevShardInfo.Unlock()
+
+	p.prevShardInfo = make(map[uint32]block.ShardData)
+	missingPreviousShardData := make(map[uint32]block.ShardData)
+
+	for _, currentShardData := range currentHeader.ShardInfo {
```

Comment:
can you extract lines 400 -> 422 into a function e.g getMissingPrevShardData

also lines

Reply:
done

Reply:
done